src/services/passwordGeneration.service.ts

```diff
@@ -157,33 +158,70 @@ export class PasswordGenerationService implements PasswordGenerationServiceAbstr
     async generatePassphrase(options: any): Promise<string> {
         const o = Object.assign({}, DefaultOptions, options);
 
+        const longestWord = WordList.EEFLongWordListOffsets.LengthLongest;
+        const shortestWord = WordList.EEFLongWordListOffsets.LengthShortest;
+
+        if (o.length == null || o.length < longestWord) {
+            o.length = longestWord;
+        }
         if (o.numWords == null || o.numWords <= 2) {
             o.numWords = DefaultOptions.numWords;
         }
-        if (o.wordSeparator == null || o.wordSeparator.length === 0 || o.wordSeparator.length > 1) {
-            o.wordSeparator = ' ';
-        }
         if (o.capitalize == null) {
             o.capitalize = false;
         }
         if (o.includeNumber == null) {
             o.includeNumber = false;
         }
+        if (o.shortWords == null) {
+            o.shortWords = false;
+        }
+        if (o.wordSeparator == null || o.wordSeparator.length === 0) {
+            o.wordSeparator = '';
+        } else if (o.wordSeparator.length > 1) {
+            o.wordSeparator = ' ';
+        }
 
-        const listLength = EEFLongWordList.length - 1;
-        const wordList = new Array(o.numWords);
-        for (let i = 0; i < o.numWords; i++) {
-            const wordIndex = await this.cryptoService.randomNumber(0, listLength);
-            if (o.capitalize) {
-                wordList[i] = this.capitalize(EEFLongWordList[wordIndex]);
-            } else {
-                wordList[i] = EEFLongWordList[wordIndex];
+        // Words with up to 5 chars are considered short
+        const listLength = o.shortWords ? WordList.EEFLongWordListOffsets.Last_5_chars :
+                                          WordList.EEFLongWordListOffsets.Last;
+
+        const wordList = new Array<string>();
+
+        if (o.type === 'passphrase_limited') { // by char limiting
```

Comment:
Could this just be an option of type passphrase? This doesn't seem like it should be a whole new type.